src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -1961,7 +1965,9 @@ static apt_bool_t speech_on_channel_add(mrcp_application_t *application, mrcp_se
 	return TRUE;
 
 error:
-	if (schannel) {
+	if (schannel && schannel->channel_destroyed) {
```

Comment:
I would suggest to check channel_destroyed inside if(schannel) which is something like this
```
if (schanne) {
   if(schannel->channel_destroyed) {
     ...
   } else {
     ...
   }
}
```
This is for readability and in the future if we need to add another validation we can just add it under if(schannel) to avoid redundant if condition that check schannel is not NULL